.github/workflows/parse_logs.py

```diff
@@ -1,21 +1,54 @@
 # type: ignore
+import argparse
+import itertools
 import pathlib
+import textwrap
 
-files = pathlib.Path("logs").rglob("**/*-log")
-files = sorted(filter(lambda x: x.is_file(), files))
+parser = argparse.ArgumentParser()
+parser.add_argument("filepaths", nargs="+", type=pathlib.Path)
+args = parser.parse_args()
+
+filepaths = sorted(p for p in args.filepaths if p.is_file())
+
+
+def extract_short_test_summary_info(path):
+    with open(path) as f:
+        lines = (line.rstrip() for line in f)
+        up_to_start_of_section = itertools.dropwhile(
+            lambda l: "=== short test summary info ===" not in l,
+            lines,
+        )
+        up_to_section_content = itertools.islice(up_to_start_of_section, 1, None)
+        section_content = itertools.takewhile(
+            lambda l: l.startswith("FAILED"), up_to_section_content
+        )
+        content = "\n".join(section_content)
+
+    return content
+
+
+def format_log_message(path):
+    py_version = path.name.split("-")[1]
+    summary = f"Python {py_version} Test Summary Info"
+    data = extract_short_test_summary_info(path)
```

Comment:
Not worth changing for this small case, but in general I find passing file-like objects (rather than string paths) more flexible as an interface — file-like objects are easy to mock / test, can be in-memory / remote storage, etc.

Reply:
thanks, that makes sense. In this case, though, we don't even need a file-like: a iterable should be enough.